packages/plugin-typescript/sources/index.ts

```diff
@@ -32,17 +43,73 @@ const afterWorkspaceDependencyAddition = async (
 
   const selected = nonNullSuggestions[0].descriptor;
   if (selected === null)
-    return;
+   return;
+
+  if (preferInteractive) {
+    const prompt = inquirer.createPromptModule();
+    const dependencyName = getDependencyName(descriptor);
+
+    const result = await prompt({
+      type: `confirm`,
+      name: `confirmed`,
+      message: `Do you want to install @types for ${dependencyName}?`,
+    });
+
+    // @ts-ignore
+    if (!result.confirmed) {
+      return;
+    }
+  }
 
   workspace.manifest[target].set(
     selected.identHash,
     selected,
   );
 };
 
+const afterWorkspaceDependencyRemoval = async (
+  workspace: Workspace,
+  dependencyTarget: suggestUtils.Target,
+  descriptor: Descriptor,
+) => {
+  if (descriptor.scope === `types`)
+    return;
+
+  const project = workspace.project;
+  const configuration = project.configuration;
+  const preferInteractive = configuration.get(`preferInteractive`);
+  const target = suggestUtils.Target.DEVELOPMENT;
+  const typesName = getTypesName(descriptor);
+
+  const ident = structUtils.makeIdent(`types`, typesName);
```

Comment:
Not sure if I should use `makeIndent` or something like `structUtils.parseIdent(typesName)`.

Reply:
Better to use `makeIdent` for names you generate; `parseIdent` is mostly meant to be used with user input (like CLI parameters, for example).